framework/include/materials/MaterialData.h

```diff
@@ -155,8 +158,13 @@ class MaterialData
   ///@}
 
   /**
-   * Resizes the properties to the specified size
-   * @param size The size of the properties to set
+   * Resizes the number of properties to the specified size (including
+   * stateful old and older properties.
+   *
+   * Warning:
+   * If size is greater than the current size, the new properties are set
+   * to NULL except * for the last element - which is set to a blank material
```

Comment:
Can you point me to where it does this?  That behavior doesn't make any sense!  If it does that... then we should create an Issue to fix it!